src/analyses/mCP.ml

```diff
@@ -249,10 +249,10 @@ struct
     let anyq = Queries.Any q in
     if M.tracing then M.traceli "query" "query %a\n" Queries.Any.pretty anyq;
     let r = match Queries.Hashtbl.find_option querycache anyq with
-      | Some r ->
+      | Some r when !lookUpCache ->
```

Comment:
This is for the new no overflow computation in the relational domain, as the old one was unsound and unused. I added a `local_no_overflow` flag in the intdomain and checked this flag after an EvalInt query. This is a kind of dirty hack, as I first tried to rewrite the Evalint to include the overflow information, but after adapting the intdomain and trying to adopt the base.ml to include the overflow information. I noticed that this would involve nearly a few thousand lines of code and deemed this not reasonable for this PR.
  
This works in most cases except for loops, where it seems like the EvalInt query is cached, which results in the EvalInt query not being newly evaluated and the flag not being set correctly. See also: https://github.com/goblint/analyzer/pull/1330#discussion_r1468277475

Reply:
This is very suspicious. We assume that queries are pure, so caching or not caching cannot make an observable difference. If it can, then something is wrong.

We don't recompute until query results stabilize, so in the presence of such impure queries we have no soundness guarantees anymore.

Reply:
This is, because for the no_overflow query uses side effects to evaluate if we have an overflow. It queries the EvalInt query and checks the side effect for an overflow. However, this does not work if the query is cached.